codegen/smithy-aws-typescript-codegen/src/main/java/software/amazon/smithy/aws/typescript/codegen/XmlShapeDeserVisitor.java

```diff
@@ -99,17 +103,23 @@ protected void deserializeStructure(GenerationContext context, StructureShape sh
             members.forEach((memberName, memberShape) -> writer.write("$L: undefined,", memberName));
         });
 
-        members.forEach((memberName, memberShape) ->
-                deserializeNamedStructureMember(context, memberName, memberShape, () -> "output"));
+        members.forEach((memberName, memberShape) -> {
+            // Grab the target shape so we can use a member deserializer on it.
+            Shape target = context.getModel().expectShape(memberShape.getTarget());
+            deserializeNamedMember(context, memberName, memberShape, () -> "output", (dataSource, visitor) -> {
+                writer.write("contents.$L = $L;", memberName, target.accept(visitor));
+            });
+        });
 
         writer.write("return contents;");
     }
 
-    void deserializeNamedStructureMember(
+    void deserializeNamedMember(
```

Comment:
This method is getting hairy, so even though it's package private, documentation is probably required at this point.